### PR TITLE
MF-830: Add SWR to active visits widget

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

This commit adds SWR-flavoured hooks to the Active Visits widget and refactors the related components to reuse these hooks for fetching data.

## Videos

The video below demonstrates:

- Revalidation in the Active Visits widget. Notice the loading spinner in the middle top of the datatable as the data is revalidated on focus.
- Data fetching in the Visit Detail component. The component loads data for the first time when the row is expanded. Note how it doesn't need to rerender when the row is collapsed and expanded again.

https://user-images.githubusercontent.com/8509731/136191307-8869c6bd-e5ed-438b-bad6-e55e4fead7f2.mp4

## Issue

https://issues.openmrs.org/browse/MF-786

### CODE REVIEW

**Comment by denniskigen** [comment]
Note to self about this widget: the pagination logic and the carbon component imports need refactoring.

File: packages/esm-active-visits-app/src/active-visits-widget/active-visits.resource.tsx [hunk 1]
```diff
@@ -1,23 +1,61 @@
...
+  };
+
+  const mapVisitProperties = (visit: Visit): ActiveVisit => ({
+    age: visit?.patient?.person?.age,
+    id: visit.uuid,
+    idNumber: visit?.patient?.identifiers[0]?.identifier,
+    gender: visit?.patient?.person?.gender,
+    name: visit?.patient?.person?.display,
+    patientUuid: visit?.patient?.uuid,
+    visitStartTime: formatDatetime(visit.startDate),
```
Comment (brandones) [inline packages/esm-active-visits-app/src/active-visits-widget/active-visits.resource.tsx:48]: Ideally, the `formatDatetime` call should happen in the render block. We should try to keep dates in canonical formats until the absolute last moment of code execution, at which point we format them for display. This reduces the chance that someone will attempt e.g. to do math on a display date.

**Review by brandones — APPROVED** [review]
Looks good, thanks @denniskigen ! Just one bit of philosophical musing on the nature of dates, feel free to ignore it or incorporate it as you see fit.

**Comment by denniskigen** [comment]
Cheers @brandones, I've added a commit that incorporates those changes.

**Pull request merged**